zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -55,84 +68,189 @@ private ZeebeRocksDbFactory(
-          new DBOptions()
-              .setCreateMissingColumnFamilies(true)
-              .setErrorIfExists(false)
-              .setCreateIfMissing(true)
-              .setParanoidChecks(true);
+      final DBOptions dbOptions = createDefaultDbOptions(closeables);
       closeables.add(dbOptions);
 
+      final Options options = new Options(dbOptions, columnFamilyOptions);
+      closeables.add(options);
+
       db =
           ZeebeTransactionDb.openTransactionalDb(
-              dbOptions,
-              dbDirectory.getAbsolutePath(),
-              columnFamilyDescriptors,
-              closeables,
-              columnFamilyTypeClass);
+              options, pathName.getAbsolutePath(), closeables, columnFamilyTypeClass);
 
     } catch (final RocksDBException e) {
-      throw new RuntimeException("Unexpected error occurred trying to open the database", e);
+      CloseHelper.quietCloseAll(closeables);
+      throw new IllegalStateException("Unexpected error occurred trying to open the database", e);
     }
     return db;
   }
 
-  private List<ColumnFamilyDescriptor> createFamilyDescriptors(
-      final List<byte[]> columnFamilyNames, final ColumnFamilyOptions columnFamilyOptions) {
-    final List<ColumnFamilyDescriptor> columnFamilyDescriptors = new ArrayList<>();
+  private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
+    final Statistics statistics = new Statistics();
+    closeables.add(statistics);
+    statistics.setStatsLevel(StatsLevel.ALL);
 
-    if (columnFamilyNames != null && !columnFamilyNames.isEmpty()) {
-      for (final byte[] name : columnFamilyNames) {
-        final ColumnFamilyDescriptor columnFamilyDescriptor =
-            new ColumnFamilyDescriptor(name, columnFamilyOptions);
-        columnFamilyDescriptors.add(columnFamilyDescriptor);
-      }
-    }
-    return columnFamilyDescriptors;
+    return new DBOptions()
+        .setErrorIfExists(false)
+        .setCreateIfMissing(true)
+        .setParanoidChecks(true)
+        // 1 flush, 1 compaction
+        .setMaxBackgroundJobs(2)
+        // we only use the default CF
+        .setCreateMissingColumnFamilies(false)
+        // may not be necessary when WAL is disabled, but nevertheless recommended to avoid
+        // many small SST files
+        .setAvoidFlushDuringRecovery(true)
+        // fsync is called asynchronously once we have at least 4Mb
+        .setBytesPerSync(4 * 1024 * 1024L)
+        // limit the size of the manifest (logs all operations), otherwise it will grow
+        // unbounded
+        .setMaxManifestFileSize(256 * 1024 * 1024L)
+        // speeds up opening the DB
+        .setSkipStatsUpdateOnDbOpen(true)
+        // keep 1 hour of logs - completely arbitrary. we should keep what we think would be
+        // a good balance between useful for performance and small for replication
+        .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
+        .setKeepLogFileNum(2)
```

Comment:
Just a thought - if we don't disable the WAL for now, we might still want to tune it by limiting its size, reusing WALs, how many to reuse, etc.

I guess that's out of scope now, I can write that on the other issue

Reply:
Please create a follow up issue for that yes.